### PR TITLE
Add option to show terms upon update

### PR DESCRIPTION
**Pull request by chartgerink**
This PR adds the option to hardcode reappearance of the terms when updating the application.

Specifically, whenever the terms are updated before release, the `welcome` and `terms` storevalue should be flipped to true. In the PR they are set to `false` given that the terms have not yet been updated in `main`. This will then show the terms to users after the app is updated (plus the final welcome screen). Specifically, it is this tidbit that needs to be changed in `./index.js`: 
```
if (!(app.getVersion() === store.get('lastVersion'))) {
  store.set('welcome', false)
  store.set('terms', false)
}
```

I implemented it by skipping all the pages between the terms and the final welcome screen. This is due to the welcome screens at the moment not being designed for single display (per [this comment](https://github.com/hypergraph-xyz/desktop/pull/350#issuecomment-773917757)). This was my inelegant approach to be able to add it at the moment, but we might want to reconsider the way the welcome screens are setup in the future if this proves too clunky.

### CODE REVIEW

File: app/components/welcome/welcome.js [hunk 1]
```diff
@@ -60,107 +60,134 @@ const ButtonGroup = styled.div`
...
-            'analytics',
-            Boolean(data.get('analytics'))
-          )
-          await ipcRenderer.invoke('setStoreValue', 'chatra', true)
-          next()
-        }}
-      >
-        <ButtonGroup>
+  ({ next }) => {
+    return (
```
Comment (juliangruber) [inline app/components/welcome/welcome.js:64]: unnecessary syntax change

File: app/components/welcome/welcome.js [hunk 2]
```diff
@@ -60,107 +60,134 @@ const ButtonGroup = styled.div`
...
+            await ipcRenderer.invoke(
+              'setStoreValue',
+              'analytics',
+              Boolean(data.get('analytics'))
+            )
+            await ipcRenderer.invoke('setStoreValue', 'chatra', true)
+            await ipcRenderer.invoke('setStoreValue', 'terms', false)
+            await ipcRenderer.invoke(
+              'setStoreValue',
+              'lastVersion',
```
Comment (juliangruber) [inline app/components/welcome/welcome.js:80]: not clear which version this is, what about `'currentAppVersion'` or `'lastInstalledAppVersion'`?

File: app/components/welcome/welcome.js [hunk 3]
```diff
@@ -60,107 +60,134 @@ const ButtonGroup = styled.div`
...
+          onSubmit={async ev => {
+            ev.preventDefault()
+            const data = new FormData(ev.target)
+            await ipcRenderer.invoke(
+              'setStoreValue',
+              'analytics',
+              Boolean(data.get('analytics'))
+            )
+            await ipcRenderer.invoke('setStoreValue', 'chatra', true)
+            await ipcRenderer.invoke('setStoreValue', 'terms', false)
```
Comment (juliangruber) [inline app/components/welcome/welcome.js:77]: Why do we set `terms` to `false` here, right after the user saw them? I didn't get that part in the PR description

File: app/components/welcome/welcome.js [hunk 4]
```diff
@@ -60,107 +60,134 @@ const ButtonGroup = styled.div`
...
+            </Button>
+            <Checkbox name='analytics' />
+            Allow analytics to be sent to Liberate Science GmbH
+          </ButtonGroup>
+        </Form>
+      </>
+    )
+  },
+  ({ page, skip, next }) => {
+    if (skip) next()
```
Comment (juliangruber) [inline app/components/welcome/welcome.js:99]: Instead of each slide manually skipping, the slide manager can simply modify the current page index

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: app/components/welcome/welcome.js (hunk 2 above)
Comment (chartgerink) [reply app/components/welcome/welcome.js:80]: Good point. `lastInstalledAppVersion` is the best depiction. I updated accordingly :)

File: app/components/welcome/welcome.js (hunk 3 above)
Comment (chartgerink) [reply app/components/welcome/welcome.js:77]: Hm, the intent is to update it to `false` after the person accepts the terms. `terms` is meant as `showTerms`, which might be the confusing part? I based it off the `welcome` store that functioned more as `showWelcome` too.

File: app/components/welcome/welcome.js (hunk 4 above)
Comment (chartgerink) [reply app/components/welcome/welcome.js:99]: lol indeed. 👍

File: app/components/welcome/welcome.js (hunk 3 above)
Comment (juliangruber) [reply app/components/welcome/welcome.js]: ah right. Yeah I think that's exactly it!

Comment (chartgerink) [reply app/components/welcome/welcome.js]: Updated the store names to be more verbose :)

**Pull request merged**